### PR TITLE
Added files that will be needed to become an official

### PR DESCRIPTION
**Pull request by dpwspoon**
@vjwang Please have some one from documentation review documentation files
@robinzimmermann Please review documentation files

### CODE REVIEW

**Comment by dpwspoon** [comment]
For reference: these files will mirror the public files @ 

https://github.com/docker-library/official-images/tree/master/library
and
https://github.com/docker-library/docs/tree/master/nginx

**Comment by pmoskovi** [comment]
I would avoid using the term "Gateway" in standalone mode. I'd always qualify it: Kaazing Gateway (even if it's capitalized, it makes it easier for the reader).
E.g.:
"The Gateway is a network gateway created to provide"
suggestion:
"The Kaazing Gateway is a network gateway created to provide"

**Comment by pmoskovi** [comment]
"protocol elevation"
If I didn't know what you meant, I'm not sure the term "protocol elevation" would help me...

**Comment by vjwang** [comment]
@michaelcretzman Please review the documentation files.

**Comment by pmoskovi** [comment]
"It is designed to provide scalable and secure bidirectional event-based communication over the web; on every platform, browser, and device."
Regarding the last part: I'd probably give examples, or list the supported platforms. Also, putting it this way may raise questions about the nature of the platform. "Do they mean CPU type/manufacturer platform, operating system platform, etc.?"

**Comment by pmoskovi** [comment]
Under the paragraph, titled "What is the Kaazing Gateway?" the logo image is broken, and am not sure that the URL will be valid in the end. Can you double-check?

**Comment by pmoskovi** [comment]
"up and running" and "configuration"
Are these "titles" part of a standard template? If not, shouldn't we have more descriptive and capitalized titles?

**Comment by pmoskovi** [comment]
"websocket"
should read:
"WebSocket"
Note: camel capitalization

**Comment by pmoskovi** [comment]
"By default the gateway runs a websocket echo service from websocket.org"
What do you mean by: "from websocket.org"?
Also, I'd add a period or a colon to the end of the sentence.

**Comment by pmoskovi** [comment]
"Note: this assumes that somehostname is resolvable from your browser, you may need to add and etc/hosts entry for somehostname on your dockerhost ip."
Not sure what our doc standards are @michaelcretzman, but I suggest we use some sort of highlighting for "somehostname" and "etc/hosts".

**Comment by pmoskovi** [comment]
"To launch a container with a specific config you can do the following"
I'd add a colon to the end of this sentence.

**Comment by dpwspoon** [comment]
"up and running" and "configuration" are not standard.  Happy to change them

**Comment by dpwspoon** [comment]
The log will be fixed when it is deployed

**Comment by dpwspoon** [comment]
The "protocol elevation" and introduction are taken from https://github.com/kaazing/gateway README, we should update that as well.  @pmoskovi can you give a better intro on what the Gateway is

**Comment by dpwspoon** [comment]
Thanks @pmoskovi

**Comment by dpwspoon** [comment]
Updated based on initial review.  Ping @michaelcretzman

File: official.repo/kaazing-gateway.docs/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# Supported tags and respective `Dockerfile` links
+
+-	[`latest`, `5.0.1.21`, (*Dockerfile*)](https://github.com/kaazing/gateway.docker/blob/master/Dockerfile)
+
+For more information about this image and its history, please see the [relevant manifest file (`library/kaazing-gateway`)](https://github.com/docker-library/official-images/blob/master/library/kaazing-gateway) in the [`docker-library/official-images` GitHub repo](https://github.com/docker-library/official-images).
+
+# What is the Kaazing Gateway?
```
Comment (michaelcretzman) [inline official.repo/kaazing-gateway.docs/README.md]: KAAZING Gateway. We upper-case KAAZING.

File: official.repo/kaazing-gateway.docs/README.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+The Gateway is a network gateway created to provide a single access point for real-time web based protocol elevation that supports load balancing, clustering, and security management. It is designed to provide scalable and secure bidirectional event-based communication over the web; on every platform, browser, and device.
```
Comment (michaelcretzman) [inline official.repo/kaazing-gateway.docs/README.md]: How about: 
The Gateway is a network gateway that provides a single access point for real-time, Web based protocol translation. The Gateway provides load balancing, clustering, and advanced security management. It enables scalable and secure bidirectional event-based communication over the Web, on every platform, browser, and device.

File: official.repo/kaazing-gateway.docs/README.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+![logo](https://raw.githubusercontent.com/docker-library/docs/master/kaazing-gateway/logo.png)
+
+# How to use this image
+
+## up and running
```
Comment (michaelcretzman) [inline official.repo/kaazing-gateway.docs/README.md]: Why the lower-case title? If there's no reason or convention that you are following, can we use title-case? Up and Running?

File: official.repo/kaazing-gateway.docs/README.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+By default the gateway runs a websocket echo service from [websocket.org](https://www.websocket.org/echo.html)
```
Comment (michaelcretzman) [inline official.repo/kaazing-gateway.docs/README.md]: How about: An instance of the Gateway runs as a publicly available WebSocket Echo service at [websocket.org](https://www.websocket.org/echo.html).

File: official.repo/kaazing-gateway.docs/README.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+    docker run --name some-kaazing-gateway -h somehostname -d -p 8000:8000 kaazing-gateway
+
+You should then be able to connect to ws://somehostname:8000 from the [websocket echo test](https://www.websocket.org/echo.html).
```
Comment (michaelcretzman) [inline official.repo/kaazing-gateway.docs/README.md]: How about: You should then be able to connect to ws://somehostname:8000 from the [WebSocket Echo test](https://www.websocket.org/echo.html).

File: official.repo/kaazing-gateway.docs/README.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+Note: this assumes that somehostname is resolvable from your browser, you may need to add and etc/hosts entry for somehostname on your dockerhost ip.
```
Comment (michaelcretzman) [inline official.repo/kaazing-gateway.docs/README.md]: How about: Note: this scenario assumes that somehostname is resolvable by your Web browser. You might need to add an entry to etc/hosts that maps somehostname to your dockerhost IP address.

File: official.repo/kaazing-gateway.docs/README.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+## configuration
```
Comment (michaelcretzman) [inline official.repo/kaazing-gateway.docs/README.md]: Title-case: Configuration

File: official.repo/kaazing-gateway.docs/README.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+To launch a container with a specific config you can do the following
```
Comment (michaelcretzman) [inline official.repo/kaazing-gateway.docs/README.md]: To launch a docker container with a specific Gateway configuration file, do the following

File: official.repo/kaazing-gateway.docs/README.md [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+
+	docker run --name some-kaazing-gateway -v /some/gateway-config.xml:/kaazing-gateway/conf/gateway-config.xml:ro -d kaazing-gateway
+
+For information on the syntax of the Kaazing Gateway configuration files, see [the official documentation](http://developer.kaazing.com/documentation/5.0/index.html) (specifically the [Configuration Guide](http://developer.kaazing.com/documentation/5.0/admin-reference/r_conf_elementindex.html)).
```
Comment (michaelcretzman) [inline official.repo/kaazing-gateway.docs/README.md]: KAAZING Gateway

File: official.repo/kaazing-gateway.docs/README.md [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+
+If you wish to adapt the default configuration, use something like the following to copy it from a running Kaazing Gateway container:
```
Comment (michaelcretzman) [inline official.repo/kaazing-gateway.docs/README.md]: If you wish to adapt the default Gateway configuration file, you can use a command such as the  following to copy the file from a running Kaazing Gateway container:

File: official.repo/kaazing-gateway.docs/README.md [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+
+## Issues
+
+If you have any problems with or questions about this image, please contact us through a [GitHub issue](https://github.com/kaazing/gateway.docker/issues).
+
+You can also reach many of the official image maintainers via the `#docker-library` IRC channel on [Freenode](https://freenode.net).
+
+## Contributing
+
+You are invited to contribute new features, fixes, or updates, large or small; we are always thrilled to receive pull requests, and do our best to process them as fast as we can.
```
Comment (michaelcretzman) [inline official.repo/kaazing-gateway.docs/README.md]: You are invited to contribute new features, fixes, or updates, large or small. We are always thrilled to receive pull requests, and do our best to process them as fast as we can.

**Comment by dpwspoon** [comment]
This was merged to the official docker.  Thanks for the feedback

**Pull request closed without merging**